async/logic/Async4BackendRest_CallbackAPI_TopicCheckLogic.java

```diff
@@ -0,0 +1,343 @@
+		executiontime.delete(0, executiontime.length());
+    	
+	    // 接続時の設定値を Properties インスタンスとして構築する
+	    Properties properties = new Properties();
+	    // 接続先 Kafka ノード
+	    properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "asynckafka1:9092");
+	    // Consumer を識別するための group id
+	    properties.put(ConsumerConfig.GROUP_ID_CONFIG, "java-consumer-group");
+	    // 読み取り位置である offset が未指定だった場合に、Kafka 上に最も早く置かれたメッセージを読み取る
+	    properties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+	    // 読み取り後に、読み取り位置である offset を自動で更新する
+	    properties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+	    // 最大1つのメッセージを取得する。
+	    properties.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 1);
+	 
+	    // Consumer を構築する
+	    KafkaConsumer<String, String> consumer =
+	        new KafkaConsumer<String, String>(properties, new StringDeserializer(), new StringDeserializer());
+	 
+	    // Consumer をトピックに割り当てる
+	    consumer.subscribe(Arrays.asList("Async4BackendRest_CallbackAPI_Topic"));
+	 
+	    try {
+	      while (true) {
+	        // メッセージをとりだす
+	        ConsumerRecords<String, String> records = consumer.poll(Duration.ofSeconds(60l));
+	        Date date = new Date();
+    		SimpleDateFormat dateformat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+    		String strDate = dateformat.format(date);
+	        
+	        // メッセージが無かったらそのまま抜ける
+			if(0 ==  records.count()) {
+				consumer.close();
+				return Response.status(Response.Status.CREATED.getStatusCode()).entity(response).build();
+			}
+	        
+	        // とりだしたメッセージを表示する
+	        for (ConsumerRecord<String, String> record : records) {
+	          receiveMessage = record.value();
+			  System.out.println(String.format("*record* %s:%s", record.offset(), record.value()));
+	        }
+			destinationResponseBody.toString().replaceAll("\"", "\\\\\"");
+	        // メッセージの読み取り位置である offset を、最後に poll() した位置で(同期処理で)更新する
+			try{
+			consumer.commitSync();
+			JSONObject jsonObject = new JSONObject(receiveMessage);
+			
+			destinationStatusCode.append(jsonObject.getString("destinationStatusCode"));
+
+			destinationResponseBody.append(jsonObject.getString("destinationResponseBody"));
+			
+			callbackUri.append(jsonObject.getString("callbackUri"));
+
+			destinationresult.append(jsonObject.getString("destinationresult"));
+
+	        transactionid.append(jsonObject.getString("transactionid"));
+
+			asyncid.append(jsonObject.getString("asyncid"));
+
+			executiontime.append(jsonObject.getString("executiontimes"));
+			}catch(CommitFailedException e){
+				e.printStackTrace();
+			}
+			return Response.status(Response.Status.CREATED.getStatusCode()).entity(response).build();
+	      }
+	    } catch(KafkaException e) {
+			e.printStackTrace();
+			System.out.println(e);
+	    	return GeneralErrorRes(ErrorResponseEnum.E500);
+	    } finally {
+	      consumer.close();
+	    }
+	  }
+
+	private Response ExecuteCallBackAPI(StringBuilder callbackUri,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 76 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.